File: Javascript_sol/Arrays_hashing/Leetcode_1832.js
/**
 * @param {string} sentence
 * @return {boolean}
 */
// Not the intended solution 
// Time complexity O(N) Space complexity O(set(N))
 var checkIfPangram = function(sentence) {
  if (new Set(sentence).size === 26) {
      return true
  };
  return false
};



/*
// Time complexity O(N + set(N)) Space complexity O(set(N))
 var checkIfPangram = function(sentence) {
  var hmap = {};
  Array.from(sentence).forEach((letter) => {
     if (hmap.hasOwnProperty(letter) === false) {
         hmap[letter] = 1
     };
  });
  
  if (Object.keys(hmap).length === 26) {
      return true
  }
  return false
};
*/

File: Python_sol/Arrays_hashing/Leetcode_1832.py
class Solution:
    def checkIfPangram(self, sentence: str) -> bool:
        # Time complexity O(N) Space complexity O(set(N))
        if len(set(sentence)) == 26:
            return True

""" Intended Solution
class Solution:
    def checkIfPangram(self, sentence: str) -> bool:
        hmap = {}
        # Time complexity O(N + set(N)) Space complexity O(set(N))
        for letter in sentence:
            if not hmap.get(letter, 0):
                hmap[letter] = 1
        if len(hmap.keys()) == 26:
            return True
"""